protocol-spec/generic.md

```diff
@@ -134,5 +144,148 @@
 ]
 ```
 
+## VibrateCmd
+
+**Description:** Causes a toy that supports vibration to run specific vibration motors at a certain speeds. Devices with multiple vibrator features may take multiple values. The [FeatureCount](enumeration.md#messageattributes) attribute for the message in the [DeviceList](enumeration.md#devicelist)/[DeviceAdded]enumeration.md#deviceadded) message will contain that information.
+
+Since devices may differ in terms of range of vibration strengths, the values are sent as a dictionary of vibration motor indexes against floats with a range of \[0.0-1.0\] wihich will be scaled appropriately for the device.
+
+**Introduced In Version:** 1
+
+**Message Version:** 1
+
+**Fields:**
+
+* _Id_ \(unsigned int\): Message Id
+* _DeviceIndex_ \(unsigned int\): Index of device
+* _Speeds_ \(array\): Vibration speeds
+  * _Index_ \(unsigned int\): Index of vibration motor
+  * _Speed_ \(double\): Vibration speed with a range of \[0.0-1.0\]
+
+**Expected Response:**
+
+* Ok message with matching Id on successful request.
+* Error message on value or message error.
+
+**Serialization Example:**
+
+```json
+[
+  {
+    "VibrateCmd": {
+      "Id": 1,
+      "DeviceIndex": 0,
+      "Speeds": [
```

Comment:
Nit/discussion point: For generic command messages, should we just call these arrays "Commands" versus "Speeds"/"Vectors"/"etc"? I think what we have currently is ok and can probably stay, just curious on your thoughts there.

Reply:
Tempted, but because the child types are different I'm not sure it'll buy us anything.
That and commands makes me think you might be able to do more per motor than control its speed (like set a pattern?). That's not the intention here.

Reply:
Ok, yeah, that's fine then.